lib/ddtrace/sampler.rb

```diff
@@ -93,37 +93,49 @@ class PrioritySampler
     extend Forwardable
 
     def initialize(opts = {})
-      @base_sampler = opts[:base_sampler] || RateSampler.new
       @post_sampler = opts[:post_sampler] || RateByServiceSampler.new
     end
 
     def sample(span)
-      return perform_sampling(span) unless span.context
-      return sampled_by_upstream(span) if span.context.sampling_priority
-
-      perform_sampling(span).tap do |sampled|
-        span.context.sampling_priority = if sampled
-                                           Datadog::Ext::Priority::AUTO_KEEP
-                                         else
-                                           Datadog::Ext::Priority::AUTO_REJECT
-                                         end
+      # If we haven't sampled this trace yet, do so.
+      # Otherwise we want to keep whatever priority has already been assigned.
+      unless sampled_by_upstream(span)
```

Comment:
My slight preference towards guard clauses makes me suggest using
```ruby
return true `sampled_by_upstream(span)`
```
However this function is short enough that its readable either way.


Also while still having a grudge against the word `sample` in this context. 
Would it make sense to refactor this method to look something like this:

```ruby
def sample(span)
   span.sampled = true
   assign_priority(span) unless priority_assigned_upstream?(span)
  
   return true
end
```
This way it makes this patently obvious that no sampling should be done here.  - Its just a suggestion though :)

Reply:
Rewrote a lot of this... let me know if this is still relevant.